crawlfrontier/contrib/scrapy/schedulers/frontier.py

```diff
@@ -154,4 +154,12 @@ def _get_exception_code(self, exception):
             return '?'
 
     def _request_is_redirected(self, request):
-        return request.meta.get('redirect_times', 0) > 0
\ No newline at end of file
+        return request.meta.get('redirect_times', 0) > 0
+
+    def _get_overused_keys(self):
+        keys = []
+        downloader = self.crawler.engine.downloader
+        for (key, slot) in downloader.slots.iteritems():
+            if len(slot.active) / float(slot.concurrency) > 1.0:
```

Comment:
I like the idea! I was trying to achieve the same goal, by incorporating this formula here. The maximum you mentioned differs from domain to domain. For example, domain x allows us to query for 20 RPS, and domain y - 0.5 RPS. Let's assume, that we want to refine the overused slots often, and aiming for efficient download. So we want, get_next_requests to be called once per 10 secs. Therefore, maximum value for domain x would be like 200 (20 times 10), and for domain y - 5. A huge difference.

So, I think such (or similar) formula will work better here.